### ISSUE TITLE
Property with zeebe:taskHeader binding not reflected in BPMN XML if exist condition for this property

### ISSUE THREAD

**Issue by Oleksiivanov** [bpmn-io/element-templates-cli#1]
### Describe the Bug

When using a conditional for  resultExpression property with a zeebe:taskHeader binding in an element template, the expected resultExpression is not being added to the <zeebe:taskHeaders> in the generated BPMN XML for specific conditions.

### Steps to Reproduce

1. Clone the repository from the test branch: https://github.com/camunda/connectors/tree/easy_post_e2e_tests
2. go to https://github.com/camunda/connectors/blob/easy_post_e2e_tests/connectors-e2e-test/connectors-e2e-test-easy-post/src/test/java/io/camunda/connector/e2e/EasyPostTests.java and run tests with debug, putting debug mark after creating result.bpmn file , like here : 
<img width="945" alt="Screenshot 2023-11-28 at 18 54 07" src="https://github.com/bpmn-io/element-templates-cli/assets/108869886/3fdb2f41-425a-4a5a-8621-ec67c9d72261">

3. find a patch with the generated result.bpmn file: 
<img width="1700" alt="Screenshot 2023-11-28 at 18 55 21" src="https://github.com/bpmn-io/element-templates-cli/assets/108869886/6acf8ac6-d946-405c-9815-cecd751b6022">
4. open result.bpmn
5. now you can see that `resultExpression` exist in the headers in the XML file : 
<img width="815" alt="Screenshot 2023-11-28 at 18 21 12" src="https://github.com/bpmn-io/element-templates-cli/assets/108869886/a76bf6bc-ed7a-4adc-b542-b05bd131eee5">
6. go to [Easy-post-connector-template.json ](https://github.com/camunda/connectors/blob/easy_post_e2e_tests/connectors/easy-post/element-templates/easy-post-connector.json#L857) and add to resultExpression property condition : 

  ``` json
    "condition": {
        "property": "operationType",
        "oneOf": [
          "createAddress",
          "createParcel",
          "createShipment",
          "buyShipment"
        ]
      },
```
<img width="764" alt="Screenshot 2023-11-28 at 18 57 20" src="https://github.com/bpmn-io/element-templates-cli/assets/108869886/864017dd-92ac-4aa6-b8df-5365bfc8eefa">

8. run tests with debug
9. go to the newly generated result.bpmn file
10. and you see that resultExpression not exist
<img width="793" alt="Screenshot 2023-11-28 at 18 58 57" src="https://github.com/bpmn-io/element-templates-cli/assets/108869886/5eb10c25-9031-46c7-a091-c207c1df0995">

### Expected Behavior
I can use the condition for `zeebe:taskHeaders` binding in the connector template for testing;
Condition not affect resultExpression and other header properties

### Environment

- Host (Browser/Node version), if applicable: [e.g. MS Edge 18, Chrome 69, Node 10 LTS]
- OS: [e.g. Windows 7]
- Library version: [e.g. 2.0.0]

**Comment by barmac** [comment]
Can you reproduce the problem in Desktop or Web Modeler?

If the bug exists in this hackdays project only, please provide a template with which I can reproduce it.

**Comment by nikku** [comment]
@Oleksiivanov Any chance to follow-up on https://github.com/bpmn-io/element-templates-cli/issues/1#issuecomment-1831557489? How can we reproduce the issue?

**Comment by sbuettner** [comment]
@nikku As @Oleksiivanov is not available this week I will try to provide a way to reproduce it. The Easy Post Connector is making use of conditions for different operations and refers to them to have dedicated result expressions per operation:
Example:
```
{
      "label": "Result expression",
      "id": "resultExpressionGet",
      "description": "Expression to map the response into process variables. Details in the <a href=\"https://docs.camunda.io/docs/components/connectors/use-connectors/#result-expression\" target=\"_blank\">documentation</a>",
      "group": "output",
      "type": "String",
      "feel": "required",
      "value": "={trackerStatus: response.body.status}",
      "binding": {
        "type": "zeebe:taskHeader",
        "key": "resultExpression"
      },
      "condition": {
        "property": "operationType",
        "oneOf": [
          "retrieveTracker"
        ]
      }
    }
```
https://github.com/camunda/connectors/blob/0c3f4af938c3d52e9143a2ea6aa1ab48eeead636/connectors/easy-post/element-templates/easy-post-connector.json#L906

When applying a template the conditions are not considered and therefore lead to a BPMN file missing the right result expression. We thought that it might be related to the fact that these bindings are using the type `zeebe:taskHeader`.

**Comment by nikku** [comment]
@barmac Can we reproduce the issue given what Simon reported?

**Comment by barmac** [comment]
I will check this and get back to you with the results.

**Comment by barmac** [comment]
I applied the template to the Service Task in this diagram:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<bpmn:definitions xmlns:bpmn="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:modeler="http://camunda.org/schema/modeler/1.0" id="Definitions_1c27vni" targetNamespace="http://bpmn.io/schema/bpmn" exporter="Camunda Modeler" exporterVersion="5.16.0" modeler:executionPlatform="Camunda Cloud" modeler:executionPlatformVersion="8.3.0">
  <bpmn:process id="Process_0hqfcqr" isExecutable="true">
    <bpmn:task id="ServiceTask" />
  </bpmn:process>
  <bpmndi:BPMNDiagram id="BPMNDiagram_1">
    <bpmndi:BPMNPlane id="BPMNPlane_1" bpmnElement="Process_0hqfcqr">
      <bpmndi:BPMNShape id="Activity_11ibm9n_di" bpmnElement="ServiceTask">
        <dc:Bounds x="160" y="80" width="100" height="80" />
      </bpmndi:BPMNShape>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</bpmn:definitions>
```

and it resulted in:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<bpmn:definitions xmlns:bpmn="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:zeebe="http://camunda.org/schema/zeebe/1.0" xmlns:modeler="http://camunda.org/schema/modeler/1.0" id="Definitions_1c27vni" targetNamespace="http://bpmn.io/schema/bpmn" exporter="Camunda Modeler" exporterVersion="5.19.0-nightly.20231218" modeler:executionPlatform="Camunda Cloud" modeler:executionPlatformVersion="8.3.0">
  <bpmn:process id="Process_0hqfcqr" isExecutable="true">
    <bpmn:serviceTask id="ServiceTask" zeebe:modelerTemplate="io.camunda.connectors.EasyPost.v1" zeebe:modelerTemplateVersion="3" zeebe:modelerTemplateIcon="data:image/svg+xml;utf8,%3Csvg xmlns=&#39;http://www.w3.org/2000/svg&#39; xmlns:xlink=&#39;http://www.w3.org/1999/xlink&#39; version=&#39;1.1&#39; width=&#39;18&#39; height=&#39;18&#39; viewBox=&#39;0 0 1000 1000&#39; xml:space=&#39;preserve&#39;%3E%3Cdesc%3ECreated with Fabric.js 3.5.0%3C/desc%3E%3Cdefs%3E%3C/defs%3E%3Crect x=&#39;0&#39; y=&#39;0&#39; width=&#39;18&#39; height=&#39;100%25&#39; fill=&#39;%23ffffff&#39;/%3E%3Cg transform=&#39;matrix(0.2007 0 0 -0.1895 500.0727 500.0583)&#39; id=&#39;344493&#39;%3E%3Cpath style=&#39;stroke: none; stroke-width: 0; stroke-dasharray: none; stroke-linecap: butt; stroke-dashoffset: 0; stroke-linejoin: miter; stroke-miterlimit: 4; is-custom-font: none; font-file-url: none; fill: rgb(17,90,241); fill-rule: nonzero; opacity: 1;&#39; vector-effect=&#39;non-scaling-stroke&#39; transform=&#39; translate(-2549.75, -2559.5281)&#39; d=&#39;M 2475 5110 c -22 -5 -59 -17 -81 -27 c -23 -10 -470 -265 -994 -568 c -682 -394 -966 -563 -999 -595 c -54 -52 -96 -131 -111 -209 c -14 -73 -14 -2229 0 -2302 c 14 -76 54 -152 108 -207 c 37 -37 257 -168 1023 -610 c 901 -519 982 -564 1050 -578 c 74 -15 130 -13 199 8 c 19 6 474 263 1010 572 c 1044 601 1047 603 1096 710 c 37 80 45 146 42 338 l -3 176 l -980 -565 c -539 -310 -1007 -578 -1040 -595 c -70 -36 -160 -47 -221 -28 c -21 7 -265 145 -541 308 l -503 296 l 0 229 l 0 229 l 33 -20 c 17 -11 244 -142 503 -291 c 517 -298 534 -306 639 -284 c 45 10 262 132 1085 610 l 1029 598 l 1 258 l 0 258 l -22 -12 c -13 -7 -473 -271 -1023 -587 c -550 -317 -1020 -582 -1045 -589 c -60 -19 -145 -11 -202 18 c -25 13 -260 150 -522 304 l -475 280 l -1 228 l 0 228 l 23 -12 c 12 -7 229 -131 482 -277 c 253 -146 480 -274 505 -285 c 52 -23 146 -28 200 -9 c 19 7 494 279 1055 605 l 1020 594 l 3 183 c 2 140 -1 197 -13 244 c -19 74 -68 158 -117 200 c -50 44 -1962 1146 -2015 1162 c -59 18 -150 24 -198 14 z&#39; stroke-linecap=&#39;round&#39;/%3E%3C/g%3E%3C/svg%3E">
      <bpmn:extensionElements>
        <zeebe:taskDefinition type="io.camunda:http-json:1" />
        <zeebe:ioMapping>
          <zeebe:input target="operationType" />
          <zeebe:input source="basic" target="authentication.type" />
          <zeebe:input target="authentication.username" />
          <zeebe:input source="SPEC_PASSWORD_EMPTY_PATTERN" target="authentication.password" />
          <zeebe:input source="20" target="connectionTimeoutInSeconds" />
        </zeebe:ioMapping>
        <zeebe:taskHeaders />
      </bpmn:extensionElements>
    </bpmn:serviceTask>
  </bpmn:process>
  <bpmndi:BPMNDiagram id="BPMNDiagram_1">
    <bpmndi:BPMNPlane id="BPMNPlane_1" bpmnElement="Process_0hqfcqr">
      <bpmndi:BPMNShape id="Activity_1637c3y_di" bpmnElement="ServiceTask">
        <dc:Bounds x="160" y="80" width="100" height="80" />
      </bpmndi:BPMNShape>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</bpmn:definitions>
```

There is no difference between whether I use this project or Desktop Modeler.

**Comment by sbuettner** [comment]
@barmac Looks like the task headers are missing in your output XML. Please try to define a value for the result expression: https://github.com/camunda/connectors/blob/0c3f4af938c3d52e9143a2ea6aa1ab48eeead636/connectors/easy-post/element-templates/easy-post-connector.json#L906 and see whether the result gets applied correctly.

**Comment by barmac** [comment]
The headers are empty but it's the result of how the template is coded, and not the element-templates-cli. If you want to have a default value in your template you need to use the `value` field on the property:

```json
    {
      "label": "Result expression",
      "id": "resultExpression",
      "description": "Expression to map the response into process variables. Details in the <a href=\"https://docs.camunda.io/docs/components/connectors/use-connectors/#result-expression\" target=\"_blank\">documentation</a>",
      "group": "output",
      "type": "String",
      "feel": "required",
      "value": "= \"my default value\"",
      "binding": {
        "type": "zeebe:taskHeader",
        "key": "resultExpression"
      },
      "condition": {
        "property": "operationType",
        "oneOf": [
          "verifyAddresById"
        ]
      }
    }
```

**Comment by sbuettner** [comment]
Sure but the issue and please @Oleksiivanov chime in please was that even if a value was provided it wasn’t successfully applied as the result expression was missing.

**Comment by barmac** [comment]
I looked into your template and found the following:

1. The property `resultExpression` requires `authenticationType` to have value `basic` in order to be active:

```json
    {
      "id" : "resultExpression",
      "label" : "Result expression",
      "description" : "Expression with condition nad prefilled value",
      "feel" : "required",
      "group" : "output",
      "value": "={addressId:response.body.id}",
      "binding" : {
        "key" : "resultExpression",
        "type" : "zeebe:taskHeader"
      },
      "type" : "Text",
      "condition": {
        "property": "authenticationType",
        "oneOf": ["basic"]
      }
```

2. Property `authenticationType` takes `""` (empty string) as the default value (`"value": ""`):

```json
    {
      "id": "authenticationType",
      "label": "Authentication Type",
      "description": "Specify the authentication type to use.",
      "type": "Dropdown",
      "value": "",
      "optional": true,
      "choices": [
        {
          "name": "None",
          "value": ""
        },
        {
          "name": "Basic",
          "value": "basic"
        },
        {
          "name": "Bearer",
          "value": "bearer"
        }
      ],
      "binding": {
        "type": "zeebe:input",
        "name": "authentication.type"
      }
    }
```

There is also a mismatch between XML and the template you shared in https://github.com/bpmn-io/element-templates-cli/issues/1#issuecomment-1864909982.

I will come up with a simple template so that we can verify it on a small, dedicated test case.

**Comment by barmac** [comment]
It looks like there is a problem with task headers handling. The result expression is not added even if I remove the condition in this template: https://github.com/bpmn-io/element-templates-cli/blob/1-task-header/test/fixtures/templates/task-header-with-condition.json

It works on the nightly though. I'd probably need to update the `bpmn-js-element-templates` distro in this project.

**Comment by nikku** [comment]
@barmac I moved this to `ready` as it looks like this may be an easy pick.

**Comment by sbuettner** [comment]
@nikku Would be great if we could prioritize this one as its blocking our e2e tests.

**Comment by nikku** [comment]
@sbuettner This is already in our next items bucket. To speed things up you may contribute a fix yourself.

**Comment by sbuettner** [comment]
@nikku Based on the comments it wasnt clear whether this only requires and update on dependencies or actual code changes in the CLI. We can certainly update dependencies and provide a PR if thats the case.

**Comment by barmac** [comment]
I will look into this this week.

**Comment by barmac** [comment]
@sbuettner @Oleksiivanov This is now published as v0.2.1

**Comment by sbuettner** [comment]
Thanks @barmac for the support. @Oleksiivanov will test the new release.
